features/FEATURE_PSA/TARGET_TFM/src/os_wrapper_cmsis_rtos_v2.c

```diff
@@ -11,31 +11,59 @@
 
 #include <string.h>
 #include "cmsis_os2.h"
+#include "rtx_os.h"
+#include "stdlib.h"
 
 /* This is an example OS abstraction layer for CMSIS-RTOSv2 */
 
 void *os_wrapper_thread_new(const char *name, int32_t stack_size,
                             os_wrapper_thread_func func, void *arg,
                             uint32_t priority)
 {
-    osThreadAttr_t task_attribs = {.tz_module = 1};
+    osRtxThread_t *thread;
+    void *stack;
 
-    /* By default, the thread starts as osThreadDetached */
-    if (stack_size != OS_WRAPPER_DEFAULT_STACK_SIZE) {
-        task_attribs.stack_size = stack_size;
+    thread = (osRtxThread_t *)malloc(osRtxThreadCbSize);
+    if(thread == NULL){
+        return NULL;
     }
-    task_attribs.name = name;
-    task_attribs.priority = (osPriority_t) priority;
 
-    return (void *)osThreadNew(func, arg, &task_attribs);
+    stack = (void *)malloc(stack_size);
+    if(stack == NULL){
+        free(thread);
+        return NULL;
+    }
+
+    const osThreadAttr_t thread_attribs = {
+#if defined (__ARM_FEATURE_CMSE) &&  (__ARM_FEATURE_CMSE == 3U)
+        .tz_module = 1,
+#endif
+        .name = name,
+        .priority = (osPriority_t) priority,
+        .cb_mem = thread,
+        .cb_size = osRtxThreadCbSize,
+        .stack_size = stack_size,
+        .stack_mem = stack};
+
+    return (void *)osThreadNew(func, arg, &thread_attribs);
 }
 
 void *os_wrapper_semaphore_create(uint32_t max_count, uint32_t initial_count,
                                   const char *name)
 {
-    osSemaphoreAttr_t sema_attrib = {0};
+    osRtxSemaphore_t *semaphore;
 
-    sema_attrib.name = name;
+    semaphore = (osRtxSemaphore_t *)malloc(osRtxSemaphoreCbSize);
+    if(semaphore == NULL){
+        return NULL;
+    }
+
+    const osSemaphoreAttr_t sema_attrib = {
+        .name = name,
+        .attr_bits = 0,
+        .cb_mem = semaphore,
+        .cb_size = osRtxSemaphoreCbSize
+    };
```

Comment:
Consider using the Mbed OS C++ Semphore class instead.

Reply:
Fixed in a later commit.